### ISSUE TITLE
Uncaught TypeError: Expected parameter accessToken

### ISSUE THREAD

**Issue by krystlc** [chrisboakes/nuxt-contentful-demo#1]
Hey, first of all, great article, it helped a lot!
There seems to be an issue once the site is generated and statically hosted (at least in Chrome):
`Uncaught TypeError: Expected parameter accessToken`.

Steps to reproduce:
1. Deploy site on Netlify
2. Open preview in Chrome
3. Check console

It's probably because the fetch method in each page component is firing that contentful request, but the `accessToken` value isn't available because it's an `env` variable.
Any clues or hints for a solution?

**Comment by somecallmejosh** [comment]
I'm seeing the same issue.

**Comment by somecallmejosh** [comment]
@krystlc did you figure out a workaround to this issue?

**Comment by krystlc** [comment]
TLDR; no, unfortunately. I went another route: creating .json files with all the data I need at the moment of generation and having components reference that instead.

Re: this `accessToken` issue: one way to fix it is to forego the .env variables and just hard code the values. But even then, my main concern was the asyncData method firing both on generation and in a static page; I don't want to incur in unnecessary API calls for data already collected. The best way I found to generate a static (non API dependent) Nuxt site with Contentful is following this pattern: https://github.com/mciccarelli/pizza-rita/blob/master/lib/getContent.js. Basically creating .json files with the data you need and referencing that in your components.

**Comment by chrisboakes** [comment]
@krystlc @somecallmejosh thanks for the kind words and finding this issue. Apologies also for my delay in fixing it for you. I get a lot of GitHub notifications from my job and this one unfortunately got lost in the noise.

Anyway, in answer to your question. All you need to do is wrap the instantiation of the Contentful SDK in a conditional. This way, when the project needs it (on build and generate) it will read the `.env` file but for production, once the site is already generated, it'll just skip it and not throw the error. I've updated the codebase with this fix (#2).

Hope this helps.
